### PR TITLE
cli: fix parameters of get/get_shared commands

### LINKED ISSUE #103: cli: use `by_me` filter in queries instead of exposing user id [1h]
Client is using a user id to filter user deposits. Server already has an option to do that, using `by_me` param in request query. 
Needs to be changed [here](https://github.com/cernanalysispreservation/cap-client/blob/d2824dbae84ecfacbb5f413670c1b2b4ad3cfbbb/cap_client/cap_api.py#L141)

### LINKED ISSUE #104: cli: add all parameter for `get-shared` command [1h]
`cap-client get` by default returns only drafts created by a user, and by flag `all` you can see all the drafts that user has access to
Let's make a similar behaviour for `cap-client get-shared`.

You need to pass `all` param [here](https://github.com/cernanalysispreservation/cap-client/blob/d2824dbae84ecfacbb5f413670c1b2b4ad3cfbbb/cap_client/cap_api.py#L149), and then:
* if all is False - query `/records/?q=&by_me=True`
* if all is True - query `/records`

**NOTE** this method is used also for fetching a specific record (with given PID), so make sure your changes don't break this functionality.

**UPDATE** Sorry, didn't mention, need to add/update tests to check behaviour with your added parameter.

### PR DESCRIPTION
**Pull request by Lilykos**
* uses by_me in queries instead of user id
* adds 'all' parameter for get_shared
* closes #103
* closes #104 

Signed-off-by: Ilias Koutsakis <ilias.koutsakis@cern.ch>

### CODE REVIEW

File: cap_client/cap_api.py [hunk 1]
```diff
@@ -135,24 +135,25 @@ def types(self):
...
     def get(self, pid=None, all=False):
         """Retrieve one or all analyses from a user."""
-        if pid or all:
-            url = urljoin('deposits/', pid)
-        else:
-            user_id = self.me().get('id', '')
-            url = urljoin('deposits/', '?q=_deposit.created_by:{}'.format(
-                user_id))
+        url = urljoin('deposits/', pid) if pid or all \
+            else urljoin('deposits/', '?q=&by_me=True')
```
Comment (annatrz) [inline cap_client/cap_api.py]: just one small thing, i know it was copied from previous code, but let's improve ;)
you don't need `urljoin` here, read what it does [here](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin)

File: cap_client/cap_api.py [hunk 2]
```diff
@@ -135,24 +135,25 @@ def types(self):
...
+                                      })
 
         return response if pid else response['hits']['hits']
 
-    def get_shared(self, pid=None):
+    def get_shared(self, pid=None, all=False):
         """Retrieve one or all shared analyses from a user."""
-        response = self._make_request(url=urljoin('records/', pid),
+        url = urljoin('records/', pid) if pid or all \
+            else urljoin('records/', '?q=&by_me=True')
```
Comment (annatrz) [inline cap_client/cap_api.py]: same here

File: cap_client/cap_api.py (hunk 1 above)
Comment (Lilykos) [reply cap_client/cap_api.py]: Sometimes PID is `None`, which means that either we keep the first `urljoin` (that keeps the url `deposit/`, despite the pid), or we remove it, and make the default pid field empty `default=''` in the command definition (at `cli/cli.py`), for both get and get_shared. 

Which one do you prefer? Or a third way if you have in mind.

Comment (annatrz) [reply cap_client/cap_api.py]: first urljoin is ok, only second one doesnt make sense

**Pull request merged**